src/DotNetWorker.Grpc/GrpcServiceCollectionExtensions.cs

```diff
@@ -68,9 +68,12 @@ public static IServiceCollection AddGrpc(this IServiceCollection services)
 #endif
 
             services.AddOptions<GrpcWorkerStartupOptions>()
-                .Configure<IConfiguration>((arguments, config) =>
+                .Configure<IConfiguration>((grpcWorkerStartupOption, config) =>
```

Comment:
This would be a good candidate for configuration binding:

``` CSharp
services.AddOptions<GrpcWorkerStartupOptions>()
    .BindConfiguration("Functions:Worker")
    .Configure<IConfiguration>((opt, config) =>
    {
        opt.HostEndpoint ??= // host + port check
        opt.RequestId ??= config["requestId"];
        // etc...
    });
    
```
 A drawback to be aware of is `.BindConfiguration` uses reflection, so it will have a startup impact (but how bad?). In NET8 they are introducing a source gen which will remove the reflection.

Also, what you have now is totally fine if you want to keep it that way.

Reply:
`IConfiguration.Bind` is one of the methods [we identified last year as non AOT friendly](https://github.com/Azure/azure-functions-dotnet-worker/issues/1055) when we investigated AOT readiness of dotnet worker.  May be this is something we can revisit as part of that AOT readiness effort later.